services/provider/server/storageclaim.go

```diff
@@ -0,0 +1,99 @@
+package server
+
+import (
+	"context"
+	"crypto/md5"
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+
+	ocsv1alpha1 "github.com/red-hat-storage/ocs-operator/api/v1alpha1"
+	kerrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/klog"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+type ocsStorageClassClaimManager struct {
+	client    client.Client
+	namespace string
+}
+
+func newStorageClassClaimManager(ctx context.Context, cl client.Client, namespace string) (*ocsStorageClassClaimManager, error) {
+	return &ocsStorageClassClaimManager{
+		client:    cl,
+		namespace: namespace,
+	}, nil
+}
+
+func generateHashForStorageClassClaim(storageConsumerUUID, storageClassClaimName string) string {
+	var s struct {
+		StorageConsumerUUID   string `json:"storageConsumerUUID"`
+		StorageClassClaimName string `json:"storageClassClaimName"`
+	}
+
+	s.StorageConsumerUUID = storageConsumerUUID
+	s.StorageClassClaimName = storageClassClaimName
+
+	claimName, err := json.Marshal(s)
+	if err != nil {
+		klog.Errorf("failed to generate storageClassClaim hash name for %q for consumer %q. %v", storageClassClaimName, storageConsumerUUID, err)
+		panic("failed to marshal")
+	}
+	name := md5.Sum([]byte(claimName))
+	return hex.EncodeToString(name[:16])
+}
+
+func getStorageClassClaimName(consumerUUID, storageClassClaimName string) string {
+	return fmt.Sprintf("storageclassclaim-%s", generateHashForStorageClassClaim(consumerUUID, storageClassClaimName))
+}
+
+// Create creates a new storageClassClaim resource and returns the storageClassClaim ID.
+func (c *ocsStorageClassClaimManager) Create(ctx context.Context, consumerUUID, storageClassClaimName, claimType, encryptionMethod string) (string, error) {
+	generatedClaimName := getStorageClassClaimName(consumerUUID, storageClassClaimName)
+	storageClassClaimObj := &ocsv1alpha1.StorageClassClaim{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: generatedClaimName,
+		},
```

Comment:
@umangachapagain @nb-ohad we need to make consumerUUID as part of the StorageClassClaim API which is missing.

Reply:
Why do we need it as a change to the API? the consumer object should be specified on the claim as an owner ref. 
And if we really want the UID we can just annotate the resource. 

Let's remember that StorageClassClaim should also be usable in converged mode where we do not have storage consumers.

Reply:
IMO the user who is creating should not add the ownerRef when creating the claim and Not sure adding important input as the annotation is a good idea.

Reply:
Our deletion process depends on StorageClassClaim being owned by StorageConsumer on provider cluster.
Adding explicit ownership should be fine here, since it's internal to the provider and created via automation.

Reply:
Why this should be done at the RPC level, not at the controller level?

Reply:
Two reasons:
1. we want it already set from the creation
2. No need to start a storageconsumer  full reconcile just to add an owner ref

Reply:
>IMO the user who is creating should not add the ownerRef when creating the claim and Not sure adding important input as the annotation is a good idea.

It is not the user who is adding it, it is the API server.

Reply:
done, added code to add ownerRef at the APIServer.